src/components/IKUpload/index.tsx

```diff
@@ -1,15 +1,34 @@
 import React from 'react';
-import ImageKitComponent from "../ImageKitComponent/";
+import ImageKitComponent from "../ImageKitComponent";
 import COMMON_PROPS from "../IKContext/props"
-import IK_UPLOAD_PROPS from "./props"
+import IK_UPLOAD_PROPS, { IKUploadProps } from "./props"
 
 const PROP_TYPES = {
   ...COMMON_PROPS,
   ...IK_UPLOAD_PROPS
 };
+type IKUploadState = {
+  xhr?: XMLHttpRequest;
+}
+
+class IKUpload extends ImageKitComponent<IKUploadProps> {
+  state: IKUploadState = {};
+  static propTypes = PROP_TYPES;
+  static defaultProps = {
+    useUniqueFileName: true,
+    isPrivateFile: false,
+    customCoordinates: "",
+    tags: [],
+    folder: "/",
+    responseFields: []
```

Comment:
Many props are missing. See the list of options upload API accepts.
<img width="1348" alt="image" src="https://user-images.githubusercontent.com/45416977/190081972-9b0686c9-a407-4186-8098-f8534df838f5.png">
Type definitions need to be added as well
<img width="585" alt="image" src="https://user-images.githubusercontent.com/45416977/190082023-fd3879ed-fde3-4fcb-b243-ee9cbb3d4a28.png">